src/controllers/update/beneficial.owner.type.controller.ts

```diff
@@ -8,17 +8,26 @@ import {
   BeneficialOwnerTypeKey,
   ManagingOfficerTypeChoice
 } from "../../model/beneficial.owner.type.model";
-import { BeneficialOwnerIndividualKey } from "../../model/beneficial.owner.individual.model";
+
 import { getApplicationData, setApplicationData } from "../../utils/application.data";
 import { ApplicationData } from "../../model";
-import { getCompanyPsc } from "../../service/persons.with.signficant.control.service";
 import { EntityNumberKey } from "../../model/data.types.model";
+import { ManagingOfficerKey } from "../../model/managing.officer.model";
+import { getCompanyOfficers } from "../../service/company.managing.officer.service";
+import { getCompanyPsc } from "../../service/persons.with.signficant.control.service";
+import { BeneficialOwnerIndividualKey } from "../../model/beneficial.owner.individual.model";
 
 export const get = async (req: Request, res: Response, next: NextFunction) => {
   try {
     logger.debugRequest(req, `${req.method} ${req.route.path}`);
+
     const appData: ApplicationData = getApplicationData(req.session);
 
+    // ManagingOfficerKey set when officer list is retrieved to prevent re-calling the API
+    if (appData[ManagingOfficerKey] === undefined) {
+      await retrieveManagingOfficers(req, appData);
+    }
+
     if (appData[BeneficialOwnerIndividualKey] === undefined) {
```

Comment:
Put two data fetches in same if (guarded by having none of 5 types). Alternatively put something is session to record the fetch was done but that would mean model changes (for save and resume).

Reply:
added util to look at all keys and return true if at least one is populated/not undefined